Fails with MariaDB, and my assumption was that the 4.* branch should keep compatibility with MariaDB?

Fails with MariaDB, and my assumption was that the 4.* branch should keep compatibility with MariaDB?

```
dbdimp.c: In function 'parse_params':
dbdimp.c:769:22: error: implicit declaration of function 'mysql_real_escape_string_quote'; did you mean 'mysql_real_escape_string'? [-Wimplicit-function-declaration]
  769 |               ptr += mysql_real_escape_string_quote(sock, ptr, valbuf, vallen, '\'');
      |                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                      mysql_real_escape_string
make[1]: *** [Makefile:358: dbdimp.o] Error 1
```


_Originally posted by @gregoa in https://github.com/perl5-dbi/DBD-mysql/pull/458#issuecomment-2644327954_

--- a/dbdimp.c
+++ b/dbdimp.c
@@ -766,7 +766,11 @@ static char *parse_params(
             if (!is_num)
             {
               *ptr++ = '\'';
+#if (MYSQL_VERSION_ID >= 50704) && !defined(MARIADB_BASE_VERSION)
               ptr += mysql_real_escape_string_quote(sock, ptr, valbuf, vallen, '\'');
+#else
+              ptr += mysql_real_escape_string(sock, ptr, valbuf, vallen);
+#endif
               *ptr++ = '\'';
             }
             else

